hoomd/conftest.py

```diff
@@ -106,7 +106,8 @@ def make_snapshot(particle_types=['A'], dimensions=3, d=1, L=20):
             s.configuration.box = box
             s.configuration.dimensions = dimensions
             s.particles.N = N
-            s.particles.position[:] = [[-d / 2, 0, 0], [d / 2, 0, 0]]
+            # shift particle positions slightly in z so MPI tests pass
+            s.particles.position[:] = [[-d / 2, 0, .1], [d / 2, 0, .1]]
```

Comment:
In some of the tests, we get energies of a system before running a simulation. When those tests are run with MPI, the domain decomposition splits the box above/below the z-axis. If the positions are left exactly on the boundary between domains (i.e. at (-d/2, 0, 0) and (d/2, 0, 0)) and we try to get energies before the simulation runs, the particles are not migrated properly and the test crashes with some kind of memory error.

Reply:
Ah, that is unfortunate. Maybe this is a bug we should make an issue about, though it is unlikely to get triggered in a normal simulation.

Reply:
It's definitely a corner case, but I'll make an issue about it just so it is recorded somewhere for future reference.

Reply:
FYI, this issue will not occur during normal simulations. The run() loop triggers particle migration which resolves the error. When you request potential energies *before* the simulation run, migration has not yet occurred. We can skirt this issue by calling the migration when setting a snapshot and after loading a file. However, I'm not certain this will catch all corner cases.